tuplex/core/include/physical/SortStage.h

```diff
@@ -0,0 +1,62 @@
+//
+// Created by Nathaneal Pitt on 6/2/21.
+//
+
+#ifndef TUPLEX_SORTSTAGE_H
+#define TUPLEX_SORTSTAGE_H
+
+#include "PhysicalStage.h"
+
+namespace tuplex {
+    // forward declaration of friend class
+    namespace codegen {
+        class StageBuilder;
+    }
+
+    /*!
+     * describes a stage which sorts data.
+     * execute as narrow stage with per-node/per-thread parallelism.
```

Comment:
can you detail this a bit more? I like that an explanation on what this does is added here!

Reply:
because input/output mode is memory, maybe add that a sorting stage requires ALL data to be in main-memory. (I.e., its not able to directly perform external sort)